src/main/java/com/google/api/codegen/util/NamePath.java

```diff
@@ -75,6 +76,24 @@ public NamePath append(String head) {
   }
 
   /**
+   * Create a new NamePath where each name piece now starts with an uppercase word.
+   */
+  public NamePath withUpperPieces() {
+    List<String> newPathPieces = new ArrayList<>();
+    for (String piece : pathPieces) {
+      String newPiece =
+          (piece.isEmpty())
+              ? piece
+              : new StringBuilder(piece.length())
+                  .append(Ascii.toUpperCase(piece.charAt(0)))
+                  .append(piece.substring(1))
+                  .toString();
+      newPathPieces.add(newPiece);
+    }
+    return new NamePath(newPathPieces);
+  }
```

Comment:
Agreed. I think an edge case is missed here as well. 


When using the function as `
protoTypeName = "alpha.beta_gamma.delta"
Namepath.dotted(protoTypeName).withUpperPieces().toDoubleColoned();
`
will be
"Alpha::Beta_gamma::Delta" as opposed to being "Alpha::BetaGamma::Delta"